### DISCUSSION TITLE
v2.0.0

### DISCUSSION THREAD

**Discussion by lukasl-dev** [lukasl-dev/waterlink#9] [General]
**Full Changelog**: https://github.com/lukasl-dev/waterlink/compare/v1.1.3...v2.0.0

<hr /><em>This discussion was created from the release <a href='https://github.com/lukasl-dev/waterlink/releases/tag/v2.0.0'>v2.0.0</a>.</em>

**Comment by dankochetov** [comment]
Should I migrate to version 2 if I'm currently using version 1 and it's working fine? Are there any performance improvements, or is it just the API rework?

**Reply by lukasl-dev** [reply]
Hey, the main goal of v2 was not to explicitly improve performance, but to simplify its implementation. Furthermore, Lavalink restrictions were implemented at library level, which should at least partially prevent exceptions from occurring on the Lavalink server.

In release v2, the payloads of the new `filters` opcode have already been implemented. I will soon deal with this new feature and therefore also implement it.

If you are already pleased with v1 and want to continue using it, this should not be a problem. Serious bugs can still be reported and will be fixed. The main focus, however, will be on v2.

**Reply by dankochetov** [reply]
Got it, thanks! On that note, I have a bugfix to submit for v1 :slightly_smiling_face: https://github.com/lukasl-dev/waterlink/pull/11
I've submited it earlier, but specified the target branch as `master` instead of `v1`.
